src/processing/shape/LineDrawer.java

```diff
@@ -0,0 +1,404 @@
+					LogLevel.INFO, 
+					"[#" + Thread.currentThread().getId() + "] "
+					+ "Using DDA Line Drawing Algorithm"
+				);
+				pixels = digitalDifferentialAnalyser(
+					pointA,
+					pointB,
+					intensity
+				);
+				break;
+			
+			// If Bresenham algorithm is to be used, then use it (fall through case)
+			case BRESENHAM:
+				
+			// By Default, use Bresenham Algorithm
+			default:
+				logger.log(
+					ModuleID.PROCESSING, 
+					LogLevel.INFO, 
+					"[#" + Thread.currentThread().getId() + "] "
+					+ "Using Bresenham Line Drawing Algorithm"
+				);
+				pixels = bresenhamLineDraw(
+					pointA,
+					pointB,
+					intensity
+				);
+		}
+		
+		// Return the computed pixel arraylists
+		return pixels;
+	}
+	
+	/**
+	 * Digital Differential Analser Algorithm
+	 * 
+	 * This algorithm constructs a line segment between two given
+	 * points. It approximates the continuous line segment on the
+	 * discrete surface (bitmap).
+	 * 
+	 * Algorithm Link:
+	 * 
+	 * @see <a href="https://en.wikipedia.org/wiki/
+	 * Digital_differential_analyzer_(graphics_algorithm)">
+	 * 
+	 * @param pointA First end point of the line segment
+	 * @param pointB Second end point of the line segment
+	 * @param intensity Intensity of each point on the segment
+	 * @return The arraylist of pixels of the line segment 
+	 */
+	private static ArrayList<Pixel> digitalDifferentialAnalyser(
+		Position pointA,
+	    Position pointB,
+	    Intensity intensity
+	) {
+		// Initialize arraylist of pixels
+		ArrayList<Pixel> pixels = new ArrayList<Pixel>();
+		
+		// Compute difference in row and column coordinates
+		int dr = pointB.r - pointA.r;
+		int dc = pointB.c - pointA.c;
+		
+		// Compute number of steps to be taken
+		int steps = Math.max(Math.abs(dr), Math.abs(dc));
+		
+		double rInc = dr / (double) steps; // Increment in row direction
+		double cInc = dc / (double) steps; // Increment in col direction
+		
+		// Initial Coordinates
+		double r = (double) pointA.r, c = (double) pointA.c;
+		
+		/*
+		 *  We iterate one more than the number of steps to take
+		 *  since we must also cover the initial coordinate
+		 */
+		for(int i = 0; i <= steps; i++, r += rInc, c += cInc) {
+			// Round the double position to integer
+			Position pos = new Position(
+				(int) Math.round(r), 
+				(int) Math.round(c)
+			);
+			
+			// Add the computed pixel
+			pixels.add(new Pixel(
+				pos,
+				new Intensity(intensity)
+			));
+		}
+		
+		// Return the computed pixel arraylist
+		return pixels;
+	}
+	
+	/**
+	 * Bresenham's Line Drawing Algorithm
+	 * 
+	 * This algorithm constructs a line segment between two given
+	 * points. It approximates the continuous line segment on the
+	 * discrete surface (bitmap). It does not perform any floating
+	 * point calculations, and optimally selects points which are
+	 * close to the actual line.
+	 * 
+	 * Algorithm Link:
+	 * 
+	 * @see <a href="https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm</a>
```

Comment:
Seems like you missed to close the `a` tag and quotes.

Reply:
Fixed.